x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/installation_button.tsx

```diff
@@ -5,57 +5,214 @@
   const { status: installationStatus } = getPackageInstallStatus(name);
 
+  const history = useHistory();
+  const { getPath } = useLink();
+
   const isInstalling = installationStatus === InstallStatus.installing;
   const isRemoving = installationStatus === InstallStatus.uninstalling;
   const isInstalled = installationStatus === InstallStatus.installed;
   const showUninstallButton = isInstalled || isRemoving;
-  const [isModalVisible, setModalVisible] = useState<boolean>(false);
-  const toggleModal = useCallback(() => {
-    setModalVisible(!isModalVisible);
-  }, [isModalVisible]);
+  const [isInstallModalVisible, setIsInstallModalVisible] = useState<boolean>(false);
+  const [isUpdateModalVisible, setIsUpdateModalVisible] = useState<boolean>(false);
+  const [upgradePackagePolicies, setUpgradePackagePolicies] = useState<boolean>(true);
+  const [agentPolicyData, setAgentPolicyData] = useState<GetAgentPoliciesResponse | null>();
+
+  useEffect(() => {
+    const fetchAgentPolicyData = async () => {
+      if (packagePolicyIds && packagePolicyIds.length > 0) {
+        const { data } = await sendGetAgentPolicies({
+          perPage: 1000,
+          page: 1,
+          // Fetch all agent policies that include one of the eligible package policies
+          kuery: `${AGENT_POLICY_SAVED_OBJECT_TYPE}.package_policies:${packagePolicyIds
+            .map((id) => `"${id}"`)
+            .join(' or ')}`,
+        });
+
+        setAgentPolicyData(data);
+      }
+    };
+
+    fetchAgentPolicyData();
+  }, [packagePolicyIds]);
+
+  const packagePolicyCount = useMemo(() => packagePolicyIds.length, [packagePolicyIds]);
+  const agentCount = useMemo(() => sumBy(agentPolicyData?.items, ({ agents }) => agents ?? 0), [
+    agentPolicyData,
+  ]);
+  const conflictCount = useMemo(() => dryRunData?.filter((item) => item.hasErrors).length, [
+    dryRunData,
+  ]);
+
+  const toggleInstallModal = useCallback(() => {
+    setIsInstallModalVisible(!isInstallModalVisible);
+  }, [isInstallModalVisible]);
 
   const handleClickInstall = useCallback(() => {
     installPackage({ name, version, title });
-    toggleModal();
-  }, [installPackage, name, title, toggleModal, version]);
+    toggleInstallModal();
+  }, [installPackage, name, title, toggleInstallModal, version]);
 
-  const handleClickUpdate = useCallback(() => {
-    installPackage({ name, version, title, fromUpdate: true });
+  const navigateToNewSettingsPage = useCallback(() => {
+    const settingsPath = getPath('integration_details_settings', {
+      pkgkey: `${name}-${version}`,
+    });
+    history.push(settingsPath);
+  }, [history, getPath, name, version]);
+
+  const handleClickUpdate = useCallback(async () => {
+    await installPackage({ name, version, title, fromUpdate: true });
   }, [installPackage, name, title, version]);
 
+  const handleClickUpgradePolicies = useCallback(async () => {
+    setIsUpgradingPackagePolicies(true);
```

Comment:
Should we add a `if (isUpgradingPackagePolicies) return` here to avoid double click bugs?